src/api.jl

```diff
@@ -6,7 +6,8 @@ import LAMMPS_jll
 import LAMMPS_jll: liblammps
 import MPI: MPI_Comm
 
-@cenum _LMP_DATATYPE_CONST::UInt32 begin
+@cenum _LMP_DATATYPE_CONST::Int32 begin
```

Comment:
This file is auto generated, in https://github.com/cesmix-mit/LAMMPS.jl/tree/main/res

Reply:
For some reason the generated file doesn't fully match the lammps source code. From lammps/src/library.h:

```C
enum _LMP_DATATYPE_CONST {
  LAMMPS_NONE = -1,     /*!< no data type assigned (yet) */
  LAMMPS_INT = 0,       /*!< 32-bit integer (array) */
  LAMMPS_INT_2D = 1,    /*!< two-dimensional 32-bit integer array */
  LAMMPS_DOUBLE = 2,    /*!< 64-bit double (array) */
  LAMMPS_DOUBLE_2D = 3, /*!< two-dimensional 64-bit double array */
  LAMMPS_INT64 = 4,     /*!< 64-bit integer (array) */
  LAMMPS_INT64_2D = 5,  /*!< two-dimensional 64-bit integer array */
  LAMMPS_STRING = 6     /*!< C-String */
};
```

LAMMPS_NONE is missing in the generated file. Which is problematic since it breaks

```julia
function extract_global_datatype(lmp::LMP, name)
    return API._LMP_DATATYPE_CONST(API.lammps_extract_global_datatype(lmp, name))
end
```

as `API.lammps_extract_global_datatype` returns -1 if the name is not found.

Reply:
This likely means that the file was generated from an older version of that header. So we need to update it, if you can do that in a separate PR that would be great.